python/tools/duplicate_structure/duplicate_structure.py

```diff
@@ -0,0 +1,259 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2024 Backlight
+
+import logging
+import argparse
+import itertools
+import time
+
+import ftrack_api
+import ftrack_api.attribute
+
+logging.basicConfig()
+
+session = ftrack_api.Session(auto_populate=False)
+
+
+class DuplicateStructure(object):
+    def __init__(self):
+        self.logger = logging.getLogger('com.ftrack.recipes.tools.duplicate_structure')
+        self.logger.setLevel(logging.INFO)
+
+        self.parser = self.args = None
```

Comment:
I would split the argument parsing from the actual code .
as in (pseudocode): 
```

args = parsearguments()
class = MyClass()
class.process(arg1, arg2)

```
as it makes it easier to read.

Reply:
Sure, copied this approach from the manage project schemas script but happy to change to this instead.